Cabal/Distribution/Simple.hs

```diff
@@ -524,9 +524,9 @@ sanityCheckHookedBuildInfo verbosity
       ++ "but the package does not have a library."
 
 sanityCheckHookedBuildInfo verbosity pkg_descr (_, hookExes)
-    | not (null nonExistant)
+    | (exe1:_) <- nonExistant
```

Comment:
this parens are unnecessary, i.e.

```haskell
    | exe1 : _ <- ...
    = ...
```

(then you need to add spaces around `:`, otherwise it will look weird :)